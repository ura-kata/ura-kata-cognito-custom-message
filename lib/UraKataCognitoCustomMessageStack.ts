import * as cdk from '@aws-cdk/core';
import * as dotenv from 'dotenv';
import { UraKataCognitoCustomMessageFunction } from './UraKataCognitoCustomMessageFunction';
import * as path from 'path';

dotenv.config();

export class UraKataCognitoCustomMessageStack extends cdk.Stack {
  constructor(scope: cdk.Construct, id: string, props?: cdk.StackProps) {
    super(scope, id, props);

    new UraKataCognitoCustomMessageFunction(
      this,
      'ura-kata-cognito-custom-message',
      path.join(__dirname, '../lib/handler.ts'),
      'ura-kata-cognito-custom-message',
      {
        URA_KATA_VALIDATION_URL: process.env.URA_KATA_VALIDATION_URL as string,
      }
    );
  }
}
