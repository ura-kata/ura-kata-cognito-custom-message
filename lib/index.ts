import { CognitoCustomMessageEvent } from '../@types/CognitoCustomMessageEvent';

const validationUrl = process.env.URA_KATA_VALIDATION_URL as string;

export async function handler(
  event: CognitoCustomMessageEvent,
  context: any
): Promise<any> {
  if (event.triggerSource === 'CustomMessage_SignUp') {
    event.response.emailSubject = 'Ura-Kata ユーザー登録確認';
    const message = `${event.userName} 様<br>
<br>
Ura-Kata へのユーザー登録をお申込み頂きありがとうございます!!<br>
登録を完了するためには以下のURLにアクセスしてください。<br>
<br>
${validationUrl}?user_name=${event.userName}&confirmation_code=${event.request.codeParameter}<br>
<br>
※このメールに心当たりのない方は破棄してください。<br>
※このメールは送信専用です。ご返信頂いても対応できません。<br>
<br>
https://ura-kata.org/
`;
    event.response.emailMessage = message;
    event.response.smsMessage = message;
  }
  return event;
}
