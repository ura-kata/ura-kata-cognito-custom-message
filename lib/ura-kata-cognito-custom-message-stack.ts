import * as cdk from '@aws-cdk/core';
import { NodejsFunction } from '@aws-cdk/aws-lambda-nodejs';
import { Runtime } from '@aws-cdk/aws-lambda';
import * as dotenv from 'dotenv';

dotenv.config();

export class UraKataCognitoCustomMessageStack extends cdk.Stack {
  constructor(
    scope: cdk.Construct,
    id: string,
    props?: cdk.StackProps,
    functionEntry?: string
  ) {
    super(scope, id, props);

    new NodejsFunction(this, 'ura-kata-cognito-custom-message', {
      entry: functionEntry,
      functionName: 'ura-kata-cognito-custom-message',
      runtime: Runtime.NODEJS_14_X,
      environment: {
        URA_KATA_VALIDATION_URL: process.env.URA_KATA_VALIDATION_URL as string,
      },
    });
  }
}
