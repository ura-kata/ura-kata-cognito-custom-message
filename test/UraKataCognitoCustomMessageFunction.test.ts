import { Stack } from '@aws-cdk/core';
import { UraKataCognitoCustomMessageFunction } from '../lib/UraKataCognitoCustomMessageFunction';
import * as path from 'path';
import { SynthUtils } from '@aws-cdk/assert';

test('ura-kata-cognito-custom-message creates', () => {
  const stack = new Stack();
  new UraKataCognitoCustomMessageFunction(
    stack,
    'func',
    path.join(__dirname, '../lib/handler.ts'),
    'func-name'
  );
  expect(SynthUtils.toCloudFormation(stack)).toMatchSnapshot();
});
