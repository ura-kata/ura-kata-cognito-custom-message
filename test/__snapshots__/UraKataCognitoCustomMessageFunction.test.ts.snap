// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ura-kata-cognito-custom-message creates 1`] = `
Object {
  "Parameters": Object {
    "AssetParameters10d8a26cfee7504284bac5d2480a609d7e22f8a95c70a99f509c6d7189282959ArtifactHashA742444D": Object {
      "Description": "Artifact hash for asset \\"10d8a26cfee7504284bac5d2480a609d7e22f8a95c70a99f509c6d7189282959\\"",
      "Type": "String",
    },
    "AssetParameters10d8a26cfee7504284bac5d2480a609d7e22f8a95c70a99f509c6d7189282959S3Bucket67BC6A30": Object {
      "Description": "S3 bucket for asset \\"10d8a26cfee7504284bac5d2480a609d7e22f8a95c70a99f509c6d7189282959\\"",
      "Type": "String",
    },
    "AssetParameters10d8a26cfee7504284bac5d2480a609d7e22f8a95c70a99f509c6d7189282959S3VersionKey8DEEF34E": Object {
      "Description": "S3 key for asset version \\"10d8a26cfee7504284bac5d2480a609d7e22f8a95c70a99f509c6d7189282959\\"",
      "Type": "String",
    },
  },
  "Resources": Object {
    "funcC3A0C2E2": Object {
      "DependsOn": Array [
        "funcServiceRoleA96CCB44",
      ],
      "Properties": Object {
        "Code": Object {
          "S3Bucket": Object {
            "Ref": "AssetParameters10d8a26cfee7504284bac5d2480a609d7e22f8a95c70a99f509c6d7189282959S3Bucket67BC6A30",
          },
          "S3Key": Object {
            "Fn::Join": Array [
              "",
              Array [
                Object {
                  "Fn::Select": Array [
                    0,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParameters10d8a26cfee7504284bac5d2480a609d7e22f8a95c70a99f509c6d7189282959S3VersionKey8DEEF34E",
                        },
                      ],
                    },
                  ],
                },
                Object {
                  "Fn::Select": Array [
                    1,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParameters10d8a26cfee7504284bac5d2480a609d7e22f8a95c70a99f509c6d7189282959S3VersionKey8DEEF34E",
                        },
                      ],
                    },
                  ],
                },
              ],
            ],
          },
        },
        "Environment": Object {
          "Variables": Object {
            "AWS_NODEJS_CONNECTION_REUSE_ENABLED": "1",
          },
        },
        "FunctionName": "func-name",
        "Handler": "index.handler",
        "Role": Object {
          "Fn::GetAtt": Array [
            "funcServiceRoleA96CCB44",
            "Arn",
          ],
        },
        "Runtime": "nodejs14.x",
      },
      "Type": "AWS::Lambda::Function",
    },
    "funcServiceRoleA96CCB44": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
  },
}
`;
